lib/mongoid/config/introspection.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Config
+
+    # This module provides a way to inspect not only the defined configuration
+    # settings and their defaults (which are available via
+    # `Mongoid::Config.settings`), but also the documentation about them. It
+    # does this by scraping the `mongoid/config.rb` file with a regular
+    # expression to match comments with options.
+    #
+    # @api private
+    module Introspection
+      extend self
+
+      # A helper class to represent an individual option, its name, its
+      # default value, and the comment that documents it.
+      class Option
+        # The name of this option.
+        #
+        # @return [ String ] The name of the option
+        attr_reader :name
+
+        # The default value of this option.
+        #
+        # @return [ Object ] The default value of the option, typically a
+        #   String, Symbol, nil, true, or false.
+        attr_reader :default
+
+        # The comment that describes this option, as scraped from
+        # mongoid/config.rb.
+        #
+        # @return [ String ] The (possibly multi-line) comment. Each line is
+        #   prefixed with the Ruby comment character ("#").
+        attr_reader :comment
+
+        # Instantiate an option from an array of Regex captures.
+        #
+        # @param [ Array<String> ] captures The array with the Regex captures
+        #   to use to instantiate the option. The element at index 1 must be
+        #   the comment, at index 2 must be the name, and at index 3 must be
+        #   the default value.
+        #
+        # @return [ Option ] The newly instantiated Option object.
+        def self.from_captures(captures)
+          new(captures[2], captures[3], captures[1])
+        end
+
+        # Create a new Option instance with the given name, default value,
+        # and comment.
+        #
+        # @param [ String ] name The option's name.
+        # @param [ String ] default The option's default value, as a String
+        #   representing the actual Ruby value.
+        # @param [ String ] comment The multi-line comment describing the
+        #   option.
+        def initialize(name, default, comment)
+          @name, @default, @comment = name, default, unindent(comment.strip)
```

Comment:
`unindent` method calls '#strip' on its argument. Should we do it here as well?

Reply:
Good catch!